### DISCUSSION TITLE
Simplify GEDCOM by using unique tags?

### DISCUSSION THREAD

**Discussion by stevespages** [FamilySearch/GEDCOM#568] [Ideas]
Having the same tag meaning more than one thing makes the GEDCOM specification more difficult to understand and also more difficult to work with when programming than is necessary.

Why not make all the tags unique? I made a script to parse the YAML files from `gedcom.io/terms/v7` and found that there are about 26 tags which are not unique. The DATE tag can mean 5 different things depending on its superstructure. I think about 37 new tags would need to be created

You could use the current URIs as tags for those cases where currently a tag has more than one meaning.

Some tags such as INDI and other record tags are unique but still have a different URI value (eg record-INDI) from the tag itself. Would it not be an advantage in terms of simplicity to make all tags the same as their URIs?

The apparent simplicity of having a smaller number of shorter tags is outweighed by the actual increase in complexity it brings to understanding and working with the specification.

Perhaps there are some downsides I have not considered?

**Comment by dthaler** [comment]
Discussion in GEDCOM Steering Committee 19 NOV 2024:
* We agree it would make parsers simpler, as long as those parsers don't care about also parsing older versions of GEDCOM.
* We believe many implementations probably care about parsing GEDCOM 5.5.1 and FamilySearch GEDCOM 7.0 and whatever future version(s) exist.
* The current tags were generally kept for backwards compatibility with earlier versions of GEDCOM.  A future major version of GEDCOM (like a hypothetical 8.0) could break backwards compatibility if this is something implementations would want.
* We would want to hear from more implementers before deciding whether this is a good direction or not.

**Reply by stevespages** [reply]
Thanks! One thing I can imagine is that major implementers might not care about this because they probably understand the spec very well and are good at thinking about complex things. Newcomers to the field would be the ones who would benefit the most. In that respect I think my proposal would help to make the spec accessible to a greater number of people.
